zipline/algorithm.py

```diff
@@ -1066,7 +1066,13 @@ def on_dt_changed(self, dt):
 
         self.portfolio_needs_update = True
         self.account_needs_update = True
-        self.performance_needs_update = True
+
+        # Do not update performance if we are at midnight. Portfolio and
+        # Account have already been adjusted for splits
+        if not normalize_date(dt) == dt:
```

Comment:
All these flags are ripe for refactoring, as @ehebert suggested yesterday, but that's out of scope for now.

Regardless, I'll have to profile this to see if we can afford an extra 390 `normalize_date` calls per day (I think probably not).  If not, I have some ideas of how to avoid this check.